### DISCUSSION TITLE
Uncomfortable in-game userpages

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#21458] [Ideas]
- Click on profile in leaderboard leading to player's default gamemode - it's very uncomfortable as **you can't change gamemode** for view. It would be better to lead to osu profile if it was osu leaderboard, to taiko if taiko etc. Also it would be better to add buttons for gamemode change.
- You can't block users using in-game userpage. Just add this button. It's very uncomfortable to use browser for search and then blocking...

**Comment by ghost** [comment]
first suggestion is duplicate of #3606

**Reply by okolenmi** [reply]
Duplicate is only first half of first suggestion. Another half is about "change game mode automatically depending on type of leaderboard you clicked on".
